app/assets/javascripts/orders.coffee

```diff
@@ -86,6 +65,24 @@ app.controller 'CheckoutController',['$scope', 'Product', 'Order', 'Address', 'C
         total += (item.source.price * item.quantity))
       total
 
+    $scope.addToCart = (item) ->
+      existingItem = _.findWhere($scope.order.order_items, { source_id: item.id })
```

Comment:
there's no debouncing here. If I smash the add button 10 times before the promise comes back i'll end up sending up 10 requests and getting 10 of the same item... right?